python/gui/auto_generated/editorwidgets/core/qgseditorwidgetwrapper.sip.in

```diff
@@ -252,12 +275,15 @@ Is forwarded to the slot setValue()
 :param feature: The new feature
 %End
 
-    virtual void setValue( const QVariant &value ) = 0;
+    virtual void setValue( const QVariant &value, const QgsAttributeMap &additionalFieldValues = QgsAttributeMap() ) = 0;
```

Comment:
This is an api break - a custom wrapper's existing setValue override will no longer work

Reply:
What would be the preferred approach?

1. Rename the new one setValues, deprecate the old one (setValue), 
2. Make the new one with no default value for 2nd argument, deprecate old one (potentially keep the method with no default argument in the future)

If we go with option 1, can we do : 
`virtual void setValue( const QVariant &value ) {setValues( value, QgsAttributeMap() );}``
Is this considered as API break too, since `setValues` might not be implemented? Or is it ok, since `setValue` has been reimplemented?

Reply:
What I would do for now:

- Add a `private virtual updateValues` method with default implementation that calls `setValue`
- Add a `public setValues` method that calls `updateValues`
- Deprecate `setValue` and recommend the usage of the above two methods
- Add a QGIS 4 TODO to remove `setValue` 
- and to change to `virtual updateValues(...) = 0;`

Reasoning: By splitting the public consumer API (`setValues`) from the implementation/override side (`updateValues`), we allow ourselves a bit more control over how subclass implementations are called and other things (e.g. logging, emitting signals, clearing caches, ...) can be introduced later on. This approach should be the default and `public virtual` should be discouraged.